src/Maestro/tests/Scenarios/githubflow-release-pipeline-nonbatched.ps1

```diff
@@ -0,0 +1,181 @@
+    $subscriptionsToDelete += $subscriptionId
+
+    Write-Host "Cloning target repo to prepare the target branch"
+    # Clone the target repo, branch, add the new dependencies and push the branch
+    GitHub-Clone $targetRepoName
+    Git-Command $targetRepoName checkout -b $targetBranch
+
+    Write-Host "Adding dependencies to target repo"
+    # Add the foo and bar dependencies
+    try {
+        Push-Location -Path $(Get-Repo-Location $targetRepoName)
+        Darc-Command add-dependency --name 'Foo' --type product --repo $sourceRepoUri
+        Darc-Command add-dependency --name 'Bar' --type product --repo $sourceRepoUri
+    }
+    finally {
+        Pop-Location
+    }
+
+    Write-Host "Pushing branch to remote"
+    # Commit and push
+    Git-Command $targetRepoName commit -am `"Add dependencies.`"
+    Git-Command $targetRepoName push origin HEAD
+    $githubBranchesToDelete += @{ branch = $targetBranch; repo = $targetRepoName}
+
+    Write-Host "Set up build for intake into target repository"
+    $buildId = New-Build -repository $sourceRepoUri -branch $sourceBranch -commit $sourceCommit -buildNumber $sourceBuildNumber -assets $assets "true"
+    Write-Host "Created build: $buildId"
+
+    # Release Pipeline will run, and if it finishes, add the build to the channel. 
+    # This will trigger the dependency update. If we don't see the PR created after 10 attempts
+    # and the build is not in the channel, fail the test
+
+    Write-Host "Waiting on Release Pipeline https://dnceng.visualstudio.com/internal/_release?definitionId=$testReleasePipelineId to complete, and a PR to be opened in $targetRepoUri"
+    # Check that the PR was created properly. poll github 
+    $tries = 10
+    $success = $false
+    while ($tries-- -gt 0) {
+        Write-Host "Checking for PRs, ${tries} tries remaining"
+        $pullRequest = Get-GitHub-PullRequests $targetRepoName $targetBranch
+        if ($pullRequest) {
+            # Find and verify PR info
+            if ($pullRequest.Count -ne 1) {
+                throw "Unexpected number of pull requests opened."
+            }
+            $pullRequest = $pullRequest[0]
+
+            $pullRequestBaseBranch = $pullRequest.head.ref
+            $githubBranchesToDelete += @{ branch = $pullRequestBaseBranch; repo = $targetRepoName}
+            $gitHubPRsToClose += @{ number = $pullRequest.number; repo = $targetRepoName }
+
+            $expectedPRTitle = "[$targetBranch] Update dependencies from $githubTestOrg/$sourceRepoName"
+            if ($pullRequest.title -ne $expectedPRTitle) {
+                throw "Expected PR title to be $expectedPRTitle, was $($pullRequest.title)"
+            }
+            
+            # Check out the merge commit sha, then use darc to get and verify the
+            # dependencies
+            Git-Command $targetRepoName fetch
+            Git-Command $targetRepoName checkout $pullRequestBaseBranch
+
+            try {
```

Comment:
I don't see where you're checking for the AzDO release here, did I miss it? (looking for Get-AzDO-Release)

Reply:
I actually meant to delete that function, as I haven't been able to get the authentication for this API quite right.

I thought for a minute that it'd be enough to check whether the PR gets opened, but on further thought that won't be enough. I'm working on getting the auth right now, and will add a proper check to see if a release for the bar build ID was triggered.